### DISCUSSION TITLE
Add "enabled" option for useVirtualizer, useWindowVirtualizer

### DISCUSSION THREAD

**Discussion by shx-riff** [TanStack/virtual#722] [Ideas]
Hi! I have a complicated page with multiple virtualizers inside one scrollable element, and I need to switch to the window virtualizers on mobile, but share all the layout, data, etc. I can't replace the `useVirtualizer` hook with the `useWindowVirtualizer` conditionally since it's not allowed by the rules of react, so the only solution is to wrap each hook with a component, and pass a virtualizer to the render prop, context, etc., and render these components by a condition. However, the code becomes messy, and all the benefits of the hook approach are canceled. With the `enabled` option it would be possible to disable observers inside the `Virtualizer` and use both hooks in one component. It would be possible to check whether the virtualizer is enabled simply by reading the `virtualizer.options.enabled`.
Thanks for the great library

**Comment by piecyk** [comment]
What about just returning null from getScrollElement?

**Reply by shx-riff** [reply]
But there can be other observers, e.g. for measuring elements, etc. Also, I believe the disabled virtualizer should not do any calculations even if we call the provided methods. Moreover, there is no `getScrollElement` option for the `useWindowVirtualizer` hook

**Reply by piecyk** [reply]
True that, having enabled sounds good, will check the PR.
